src/AppBundle/Manager/UserManager.php

```diff
@@ -0,0 +1,289 @@
+     * @param User $user
+     * @return array
+     */
+    private function getMatchResultsdByPager(Pagerfanta $pagerfanta, User $user)
+    {
+        $datas  = [];
+        $users  = $pagerfanta->getCurrentPageResults();
+
+        foreach ($users as $auser) {
+            $currentUser    = $this->getFind($auser['id']);
+            $datas[]        = [
+                'user_id'           => $auser['id'],
+                'score'             => $auser['score'],
+                'interests'         => $this->getCategoriesExactMatchByUser($user, $currentUser),
+                'user_info'         => $currentUser->getFullName(),
+                'edit_profile_link' => $this->router->generate('admin_ajax_edit', ['id' => $auser['id']]),
+                'mark_pending_link' => $this->router->generate('admin_ajax_connection_request_mark_pending_or_unpending', ['id' => $auser['connection_request_id']]),
+                'mark_pending_label'=> ($auser['pending']? 'Remove Pending': 'Make Pending'),
+                'about'             => $currentUser->getAbout(),
+                'matches'           => $this->getExactMatchByUser($user, $currentUser),
+                'ele'               => 'ele'.$auser['id'],
+                'gender'            => ($currentUser->getGender() == $user->getGender()? 1: 0),
+                'age_diff'          => $currentUser->getAge()-$user->getAge()
+            ];
+        }
+
+        return $datas;
+    }
+
+    /**
+     * @param User $user
+     * @param User $currentUser
+     * @return array
+     */
+    private function getExactMatchByUser(User $user, User $currentUser)
+    {
+        $ageDiff    = $currentUser->getAge()-$user->getAge();
+        $matches    = [];
+
+        if ($this->isAgeDiffWithinRange($ageDiff)) {
+            $matches[] = $this->wrapSpanString($currentUser->getAge().' '.$this->translator->trans('years'));
+        } else {
+            $matches[] = $currentUser->getAge().' '.$this->translator->trans('years');
+        }
+
+        $matches[]  =  $currentUser->getCountryName();
+
+        if ($this->isUserMunicipalityMatch($user, $currentUser)) {
+            $matches[] = $this->wrapSpanString($currentUser->getMunicipality()->getName());
+        } else {
+            $matches[] = $currentUser->getMunicipality()->getName();
+        }
+
+        if ($this->isUserHasChildrenMatch($user, $currentUser)) {
+            $matches[] = $this->wrapSpanString(($currentUser->hasChildren()? $this->translator->trans('kids'): $this->translator->trans('no kids')));
+        } else {
+            $matches[] = ($currentUser->hasChildren()? $this->translator->trans('kids'): $this->translator->trans('no kids'));
+        }
+
+        return $matches;
+    }
+    /**
+     * @param User $user
+     * @return string
+     */
+    public function getCategoryNameStringByUser(User $user)
+    {
+        $categoryNames  = array_values($user->getCategoryNames());
+        if ($categoryNames > 1) {
```

Comment:
Looks like this is the only thing left. But a comment on the unresolved comments would be appreciated.